tests/TestPasskeys.cpp

```diff
@@ -0,0 +1,474 @@
+                "Y3woX2b2klG2fB2alGZHHrVg6yVEmnAii4kYSdmoWxI7SmzLftoZfCJNFPFHujx2Pbr-6dIB02sZhtncetT0cpyWobtj9r7C5dIGfm"
+                "J5n-LccP-F9gXGqtbN605VrIkC2WNztjdk3dAt5FGM_dlIwSe-vP1dKfIuNqAEbgr2IVZAUFn_ZfzUo-XbXTysksuz9JZfEopJBiUi"
+                "9tjQDNvrYQFqB6wDPqkZAomkbRCohUb3TzCg"));
+}
+
+void TestPasskeys::testCreatingAttestationObjectWithEC()
+{
+    // Predefined values for a desired outcome
+    const auto id = QString("yrzFJ5lwcpTwYMOdXSmxF5b5cYQlqBMzbbU_d-oFLO8");
+    const auto predefinedFirst = QString("BuyvNFtikWFtGVkDplAqyjHElahp5fCH5dS4Ms0Ihd0");
+    const auto predefinedSecond = QString("4u5_6Q8O6R0Hg0oDCdtCJLEL0yX_GDLhU5m3HUIE54M");
+
+    const auto publicKeyCredentialOptions = browserMessageBuilder()->getJsonObject(PublicKeyCredentialOptions.toUtf8());
+
+    auto rpIdHash = browserMessageBuilder()->getSha256HashAsBase64(QString("webauthn.io"));
+    QCOMPARE(rpIdHash, QString("dKbqkhPJnC90siSSsyDPQCYqlMGpUKA5fyklC2CEHvA"));
+
+    TestingVariables testingVariables = {id, predefinedFirst, predefinedSecond};
+    auto result = browserPasskeys()->buildAttestationObject(publicKeyCredentialOptions, "", id, testingVariables);
+    QCOMPARE(
+        QString(result.cborEncoded),
+        QString("\xA3"
+                "cfmtdnonegattStmt\xA0hauthDataX\xA4t\xA6\xEA\x92\x13\xC9\x9C/t\xB2$\x92\xB3 \xCF@&*\x94\xC1\xA9P\xA0"
+                "9\x7F)%\x0B`\x84\x1E\xF0"
+                "E\x00\x00\x00\x01\x01\x02\x03\x04\x05\x06\x07\b\x01\x02\x03\x04\x05\x06\x07\b\x00 \x8B\xB0\xCA"
+                "6\x17\xD6\xDE\x01\x11|\xEA\x94\r\xA0R\xC0\x80_\xF3r\xFBr\xB5\x02\x03:"
+                "\xBAr\x0Fi\x81\xFE\xA5\x01\x02\x03& \x01!X "
+                "e\xE2\xF2\x1F:cq\xD3G\xEA\xE0\xF7\x1F\xCF\xFA\\\xABO\xF6\x86\x88\x80\t\xAE\x81\x8BT\xB2\x9B\x15\x85~"
+                "\"X \\\x8E\x1E@\xDB\x97T-\xF8\x9B\xB0\xAD"
+                "5\xDC\x12^\xC3\x95\x05\xC6\xDF^\x03\xCB\xB4Q\x91\xFF|\xDB\x94\xB7"));
+
+    // Double check that the result can be decoded
+    BrowserCbor browserCbor;
+    auto attestationJsonObject = browserCbor.getJsonFromCborData(result.cborEncoded);
+
+    // Parse authData
+    auto authDataJsonObject = attestationJsonObject["authData"].toString();
+    auto authDataArray = browserMessageBuilder()->getArrayFromBase64(authDataJsonObject);
+    QVERIFY(authDataArray.size() >= 37);
+
+    auto authData = browserPasskeys()->parseAuthData(authDataArray);
+    auto credentialData = authData["credentialData"].toObject();
+    auto flags = authData["flags"].toObject();
+    auto publicKey = credentialData["publicKey"].toObject();
+
+    // The attestationObject should include the same ID after decoding with the response root
+    QCOMPARE(credentialData["credentialId"].toString(), QString("yrzFJ5lwcpTwYMOdXSmxF5b5cYQlqBMzbbU_d-oFLO8"));
+    QCOMPARE(authData["rpIdHash"].toString(), QString("dKbqkhPJnC90siSSsyDPQCYqlMGpUKA5fyklC2CEHvA"));
```

Comment:
AFAIK, this should now work without explicitly wrapping the literal in QString(). I think that was only a Qt 5.6 restriction.

Reply:
5.12 is the minimum requirement for these tests.

Reply:
Alright, then never mind.